chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -1659,6 +1659,14 @@ impl PeerManagerActor {
     #[perf]
     fn handle_msg_outbound_tcp_connect(&self, msg: OutboundTcpConnect, ctx: &mut Context<Self>) {
         let _d = delay_detector::DelayDetector::new(|| "outbound tcp connect".into());
+        let handshake_permit = match self.state.tier2.start_outbound(msg.peer_info.id.clone()) {
```

Comment:
maybe sink this under the `if let` to not immediately drop permit if there's no address? 

Though, doesn't really matter. I think this case is actually unreachable, and we shold add an `addr` field to  `OutboundTcpConnect` to statically prove that, which is out of scope for the PR

Reply:
done